connector/google/google.go

```diff
@@ -267,6 +267,9 @@ func (c *googleConnector) getGroups(email string) ([]string, error) {
 // sets up super user impersonation and creates an admin client for calling
 // the google admin api
 func createDirectoryService(serviceAccountFilePath string, email string) (*admin.Service, error) {
+	if len(serviceAccountFilePath) == 0 && len(email) == 0 {
```

Comment:
Can I make the suggestion that this be an or instead?
```suggestion
	if len(serviceAccountFilePath) == 0 || len(email) == 0 {
```
Also, if we are doing this, I think we should check if the directory service is `nil` instead of checking the configuration, WDYT? https://github.com/dexidp/dex/blob/40af15ffd3a4e8868475a976633db4a96783e434/connector/google/google.go#L215

Reply:
I think `&&` is better.  If you make it be an `or`, then if the user sets serviceAccountFilePath without AdminEmail, or vice versa, then the configuration will silently be accepted but ignored.  With `and`, then they will get an error (either from opening an empty filename, or from using an empty admin username)

However I agree that the other check should be for `adminSrv != nil`.  I made that change independently in #1720.

Reply:
I also suggest updating the commented-out example in `examples/config-dev.yaml` from `type: oidc` to `type: google`

Reply:
> I think && is better. If you make it be an or, then if the user sets serviceAccountFilePath without AdminEmail, or vice versa, then the configuration will silently be accepted but ignored. With and, then they will get an error (either from opening an empty filename, or from using an empty admin username)

I would expect this error to be picked up in the configuration validation on startup of the application? Though I understand your explanation, it makes sense. If we can verify that this option is being checked on startup, then that would be ideal

Reply:
I've added explicit validation in #1720, how does that look?